File: dotnet/src/CarespaceSDK/Api/AuthApi.cs
using CarespaceSDK.Http;
using CarespaceSDK.Models;

namespace CarespaceSDK.Api;

public class AuthApi : IAuthApi
{
    private readonly IHttpClient _httpClient;

    public AuthApi(IHttpClient httpClient)
    {
        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
    }

    public async Task<ApiResponse<LoginResponse>> LoginAsync(LoginRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PostAsync<ApiResponse<LoginResponse>>("/auth/login", request, cancellationToken);
    }

    public async Task<ApiResponse<LoginResponse>> LoginAsync(string email, string password, CancellationToken cancellationToken = default)
    {
        var request = new LoginRequest { Email = email, Password = password };
        return await LoginAsync(request, cancellationToken);
    }

    public async Task<ApiResponse<LoginResponse>> RefreshTokenAsync(RefreshTokenRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PostAsync<ApiResponse<LoginResponse>>("/auth/refresh", request, cancellationToken);
    }

    public async Task<ApiResponse<LoginResponse>> RefreshTokenAsync(string refreshToken, CancellationToken cancellationToken = default)
    {
        var request = new RefreshTokenRequest { RefreshToken = refreshToken };
        return await RefreshTokenAsync(request, cancellationToken);
    }

    public async Task<ApiResponse<object>> LogoutAsync(LogoutRequest? request = null, CancellationToken cancellationToken = default)
    {
        return await _httpClient.PostAsync<ApiResponse<object>>("/auth/logout", request, cancellationToken);
    }

    public async Task<ApiResponse<object>> ForgotPasswordAsync(ForgotPasswordRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PostAsync<ApiResponse<object>>("/auth/forgot-password", request, cancellationToken);
    }

    public async Task<ApiResponse<object>> ForgotPasswordAsync(string email, CancellationToken cancellationToken = default)
    {
        var request = new ForgotPasswordRequest { Email = email };
        return await ForgotPasswordAsync(request, cancellationToken);
    }

    public async Task<ApiResponse<object>> ResetPasswordAsync(ResetPasswordRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PostAsync<ApiResponse<object>>("/auth/reset-password", request, cancellationToken);
    }

    public async Task<ApiResponse<object>> ResetPasswordAsync(string token, string newPassword, CancellationToken cancellationToken = default)
    {
        var request = new ResetPasswordRequest { Token = token, NewPassword = newPassword };
        return await ResetPasswordAsync(request, cancellationToken);
    }

    public async Task<ApiResponse<object>> ChangePasswordAsync(ChangePasswordRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PostAsync<ApiResponse<object>>("/auth/change-password", request, cancellationToken);
    }

    public async Task<ApiResponse<object>> ChangePasswordAsync(string currentPassword, string newPassword, CancellationToken cancellationToken = default)
    {
        var request = new ChangePasswordRequest { CurrentPassword = currentPassword, NewPassword = newPassword };
        return await ChangePasswordAsync(request, cancellationToken);
    }
}

File: temp-repos/carespace-kotlin-sdk/carespace-sdk/src/commonMain/kotlin/com/carespace/sdk/api/AuthApi.kt
package com.carespace.sdk.api

import com.carespace.sdk.http.CarespaceHttpClient
import com.carespace.sdk.models.*

/**
 * Authentication API interface
 */
interface AuthApi {
    suspend fun login(request: LoginRequest): ApiResponse<LoginResponse>
    suspend fun login(email: String, password: String): ApiResponse<LoginResponse>
    suspend fun refreshToken(request: RefreshTokenRequest): ApiResponse<LoginResponse>
    suspend fun refreshToken(refreshToken: String): ApiResponse<LoginResponse>
    suspend fun logout(request: LogoutRequest? = null): ApiResponse<Unit>
    suspend fun forgotPassword(request: ForgotPasswordRequest): ApiResponse<Unit>
    suspend fun forgotPassword(email: String): ApiResponse<Unit>
    suspend fun resetPassword(request: ResetPasswordRequest): ApiResponse<Unit>
    suspend fun resetPassword(token: String, newPassword: String): ApiResponse<Unit>
    suspend fun changePassword(request: ChangePasswordRequest): ApiResponse<Unit>
    suspend fun changePassword(currentPassword: String, newPassword: String): ApiResponse<Unit>
}

/**
 * Authentication API implementation
 */
class AuthApiImpl(private val httpClient: CarespaceHttpClient) : AuthApi {

    override suspend fun login(request: LoginRequest): ApiResponse<LoginResponse> {
        return httpClient.post("/auth/login", request)
    }

    override suspend fun login(email: String, password: String): ApiResponse<LoginResponse> {
        return login(LoginRequest(email, password))
    }

    override suspend fun refreshToken(request: RefreshTokenRequest): ApiResponse<LoginResponse> {
        return httpClient.post("/auth/refresh", request)
    }

    override suspend fun refreshToken(refreshToken: String): ApiResponse<LoginResponse> {
        return refreshToken(RefreshTokenRequest(refreshToken))
    }

    override suspend fun logout(request: LogoutRequest?): ApiResponse<Unit> {
        return httpClient.post("/auth/logout", request)
    }

    override suspend fun forgotPassword(request: ForgotPasswordRequest): ApiResponse<Unit> {
        return httpClient.post("/auth/forgot-password", request)
    }

    override suspend fun forgotPassword(email: String): ApiResponse<Unit> {
        return forgotPassword(ForgotPasswordRequest(email))
    }

    override suspend fun resetPassword(request: ResetPasswordRequest): ApiResponse<Unit> {
        return httpClient.post("/auth/reset-password", request)
    }

    override suspend fun resetPassword(token: String, newPassword: String): ApiResponse<Unit> {
        return resetPassword(ResetPasswordRequest(token, newPassword))
    }

    override suspend fun changePassword(request: ChangePasswordRequest): ApiResponse<Unit> {
        return httpClient.post("/auth/change-password", request)
    }

    override suspend fun changePassword(currentPassword: String, newPassword: String): ApiResponse<Unit> {
        return changePassword(ChangePasswordRequest(currentPassword, newPassword))
    }
}